### DISCUSSION TITLE
Help needed for check if a price is bigger than the other

### DISCUSSION THREAD

**Discussion by ramon-godas-youtube** [colinhacks/zod#3620] [Q&A]
```js
const FormSchema = z
	.object({
	priceBeforeDiscount: z.preprocess(
			(val) => (val === '' ? 0 : Number(val)),
			z
				.number({ message: 'Ingresa un numero valido' })
				.min(1, { message: 'El precio no puede ser inferior a 1.' })
				.max(100000, { message: 'No aceptamos productos tan caros' })
		),
		priceAfterDiscount: z.preprocess(
			(val) => (val === '' ? 0 : Number(val)),
			z
				.number({ message: 'Ingresa un numero valido' })
				.min(0, { message: 'El precio no puede ser negativo.' })
		),
		}).refine(
		(data) => {
			if (data.priceAfterDiscount === 0) return true;
			return data.priceAfterDiscount <= data.priceBeforeDiscount;
		},
		{
			message:
				'El precio con descuento no puede ser superior al precio sin descuento',
			path: ['priceAfterDiscount'],
		}
	);
```
	
	
	It never triggers the refine validation always pass it as if was true once the fields have any number.

**Comment by marilari88** [comment]
hi @ramon-godas-youtube It seems that your schema is working as expected  [Link](https://zod-playground.vercel.app?appdata=N4IgzgxgFgpgtgQxALhALwHQHsBGArGCAFwApgAdAO3KIAcAnASwhgCEYAzLemAEUchYArpSLIABJgYwGWFmDAkqNFaQBuCADYBKcQF4AfOJIbN%2BvXvEByK%2BID84gAziJAOSFwcMeia3btADTKRKpowao0GJQeXj7A4nAwCggA5jASVgCSlCk8YAjiIuLRifRY4qaMACZYtgC%2B2uEhzRhwjJQkAIwB4vGJyWkZAKJm0hCM5ZTltEIwVTDiYN7i7RzeE-TiBZ0Y9Y3UzaqtCAAeXY4XFz19SfmD1q7lCCy0RAhwWGDislVCxJ-iN6UcQQBBlMB7JqBJoMZgwACCHCI3n4ghEYkkGGksnkiiaNF8On0RkJ5ksNnsThc4ncnm8hP8QQOoXxLRKsTICVuqXS1myuSSBSKJW85UqNUhzMORFa7RIjmuXIGvKsI2%2BPHGk2ms3mi2WlBgKQQREYaiwu3EDShTNUDQwPA47RgSilJCqxoQukMvVZjA4xndbyxTBYiOR9FREGEojJTl0PCIQnowKI9FmAG5WQmk8DAwhg3CwyiBFH0eIADyWPMFljsLg8SPRoiZqV1G3NChSmj9O7pVk0VWjDUTEFYXNJCCzURa746hZLTZgIS0dbcLZDwgjsDtcTzSBToi1duqWjGqASADaVlhoaRxbRoisAF1j0Q6sFtOmQAEQKZZmAUAvEBO3EUD1ThOtuD4Esm2QABOBCmTA8Db3DRt0WQTpLiZOpv2AqhkJvNhOCg9DREwi5ugIsCiKLCMYIwrCrioXCX1-bxtzHFAQAAZgwAAmPiAA4QDqIA). 
Can you write an example of data that doesn't trigger the refine validation?

**Reply by ramon-godas-youtube** [reply]
Yeah you are right it is mostly when the validation is checked since it is a multiple step form i had to check at the final step to trigger once again if not some times does not triggers if you type very quickly and had to change for it as well.


```js
.superRefine((data, ctx) => {
		if (
			data.priceAfterDiscount !== 0 &&
			data.priceAfterDiscount > data.priceBeforeDiscount
		) {
			ctx.addIssue({
				code: z.ZodIssueCode.custom,
				message:
					'El precio con descuento no puede ser superior al precio sin descuento',
				path: ['priceAfterDiscount'],
			});
		}
	});```


```
Probably my problem how i handle the form

So i don't really know why if i skip step 1 quickly does not trigger i just had to move it to the last step to make it more user friendly
